### ISSUE TITLE
"or" keyword for logical OR in Metadata.cpp

### ISSUE THREAD

**Issue by victoryc** [facebookresearch/SparseConvNet#185]
I have been trying to get this working on Windows.  While investigating one of the build issues, I noticed that in Metadata.cpp, there are several occurrences of the keyword "or" for logical or (e.g., on line 335) while the standard syntax for that in C++ is "||". I am wondering whether "or" is being defined to be "||" somewhere else and that definition happens to be not visible while building on Windows. 

Edit: Later on, I noticed occurrences of "and" (instead of &&) in NetworkInNetwork.cpp.

**Comment by bottler** [comment]
The words `and` and `or` can be allowed in `C++`. They are called *alternative* operators for the `||` and `&&` operators. If you are using the microsoft compiler you need to add `/Za` as a compiler option to allow them.

**Comment by victoryc** [comment]
Thanks!
By chance, do you know the whole list of options that need to be passed to get this library to build with Microsoft compiler? While passing '/Za' option by editing setup.py fixed those "or/"and" errors, there are other errors left. For example:

`...\lib\site-packages\torch\include\ATen/core/Dict.h(98): error C2990: 'c10::Dict': non-class template has already been declared as a class template`

**Comment by victoryc** [comment]
For the benefit of anyone else running into this: Use of '/Za' flag to fix this is not a good idea. Though it fixes the "and"/"or" issue itself, it leads to other errors. Specifically, as I found out, the "Dict.h" error I mentioned in previous message above occurs only when the '/Za' flag is used. Upon searching, I ran into the following thread where people have talked about the problems with the '/Za' flag: [](https://www.reddit.com/r/cpp/comments/76pmky/til_c_has_and_or_and_not/). Instead, a safer flag to use handle the 'and'/'or' keywords happened to be '/permissive-'.
